Obsidian/Logging/Logger.cs

```diff
@@ -48,59 +51,52 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
-                        Console.Write($"{"",1}[{Prefix}] ");
+                        prefix += $"{"",1}[{Prefix}] ";
                 }
                 else
                 {
                     if (Prefix.Length >= 12)
-                        Console.Write($"{"",1}[{Prefix}] ");
+                        prefix += $"{"",1}[{Prefix}] ";
                     else
-                        Console.Write($"[{Prefix}] ");
+                        prefix += $"[{Prefix}] ";
                 }
+                prefix.RenderColoredConsoleMessage();
+                #endregion
 
+                #region Message coloring & line break handling
                 var message = formatter(state, exception);
-                var msgLst = message.Contains("§") ? message.Split("§") : new string[] { $"r{message}" };
-                if (message[0] != '§' && msgLst.Length > 1) msgLst[0] = $"r{msgLst[0]}";
+                var msgLst = message.Contains("§") ? message.Split("§") : new string[] { $"f{message}" };
+                if (msgLst.Length > 1 && msgLst[0].Length > 0 && msgLst[0][0] != '§') msgLst[0] = $"f{msgLst[0]}";
                 foreach (var msg in msgLst)
                 {
-                    if (!string.IsNullOrEmpty(msg))
+                    if (!string.IsNullOrEmpty(msg) && msg.Length > 1)
                     {
-                        Console.ForegroundColor = msg[0].ToString().ToLower()[0] switch
+                        var colorStr = msg[0].ToString().ToLower()[0];
+                        
+                        var color = ChatColor.FromCode(colorStr).ToConsoleColor();
+                        string[] lines = msg.Contains("\n") ? msg.Split("\n").ToArray() : new string[] { msg };
+
+                        for (int i = 0; i < lines.Length; i++)
                         {
-                            '0' => ConsoleColor.Black,
-                            '1' => ConsoleColor.DarkBlue,
-                            '2' => ConsoleColor.DarkGreen,
-                            '3' => ConsoleColor.DarkCyan,
-                            '4' => ConsoleColor.DarkRed,
-                            '5' => ConsoleColor.DarkMagenta,
-                            '6' => ConsoleColor.DarkYellow,
-                            '7' => ConsoleColor.Gray,
-                            '8' => ConsoleColor.DarkGray,
-                            '9' => ConsoleColor.Blue,
-                            'a' => ConsoleColor.Green,
-                            'b' => ConsoleColor.Cyan,
-                            'c' => ConsoleColor.Red,
-                            'd' => ConsoleColor.Magenta,
-                            'e' => ConsoleColor.Yellow,
-                            'f' => ConsoleColor.White,
-                            'r' => ConsoleColor.White,
-                            _ => ConsoleColor.White
-                        };
-                        Console.Write(msg.Substring(1));
+                            if (i > 0) Console.WriteLine("");
```

Comment:
It's fine, but there is an overload of [Console.WriteLine](https://docs.microsoft.com/en-us/dotnet/api/system.console.writeline?view=netcore-3.1#System_Console_WriteLine) that takes no arguments.

Reply:
it errored to me, that it is not exists iirc

Reply:
That's very weird. Feel free to ignore my comment then.

Reply:
ohh it was with `Console.Write()`. I'll change this as well

Reply:
changed to `Console.WriteLine()`